fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Request.kt

```diff
@@ -90,10 +92,26 @@ class Request(
         return this
     }
 
+    /**
+     *  <p> Note that your value, will be converted to a String via the toString() </p>
+     *  <p> Please note that header of the same key are supported and headers with the same key
+     *  will be sent in the format of `key` : [ value,value,value ] </p>
+     * @param pairs This all the key value pair you wish to add to the headers
+     *
+     * @return the request supplied
+     *
+     * */
     fun header(vararg pairs: Pair<String, Any>?): Request {
-        pairs.forEach {
-            if (it != null)
-                headers += Pair(it.first, it.second.toString())
+        pairs.filterNotNull().forEach { (key,value) ->
+            if (!headers.containsKey(key)) {
+                headers += Pair(key, value.toString())
+            } else {
+                headers[key] =  headers.getValue(key).let {
+                    var str = it
+                    if (str[0] != '[') str = "[ $str ]"
+                     str.replace(" ]", "; $value ]")
```

Comment:
What is our stance on the `[` and `]`? According to the RFC, we don't need them?

Reply:
@kittinunf @SleeplessByte  I am happy to do whatever you feel is best, I have no strong feelings on this

Reply:
They don't work! If you send a header with `[]` it will literally produce this:

```
Cookie: [val1=a; val2=b; val3=c]
```

output the following cookies pairs:
- `[val1=a`
- `val2=b`
- `val3=c]`

Reply:
@SleeplessByte In my latest commit I have added a test to hit http and analyse what we get back. in you comment above do I need to remove the `;` and the `[]`?